gips/0082.md

```diff
@@ -0,0 +1,245 @@
+
+After collecting consumption signals, the Payment Gateway MUST sum up all what is due to the Indexer, and transfer a payment to the Indexer's wallet, using the `collect` function in the staking contract.
+
+### Current implementation and future opportunities
+
+THIS SECTION IS NON NORMATIVE
+
+An example of the payment flow today, which ends up abiding by this specification (calling `collect`) can be seen here:
+
+https://github.com/streamingfast/network-payments-cli
+
+It involves a small dance, of publishing a subgraph manifest, like:
+
+```yaml
+specVersion: 0.0.5
+description: "thegraph.market Payment Gateway usage"
+usage:
+  serviceName: substreams
+  namespace: sf.substreams.rpc.v2.Stream
+  network: mainnet
+```
+
+then allocating to it, sharing the allocation ID with the Payment Gateway, in order for the Indexer to receive payment.
+
+This flow is expected to be simplified in the future, with the Payment Gateway being able to automatically detect new Substreams services, and allocate to them, without manual intervention.
+
+Future opportunities would include self-serve APIs for Indexers to claim payment and do the dance automatically. Thus an Indexer would be able to lower the risks of not receiving payments from the Gateway.
+
+Eventually, TAP could be used to send payments alongside blockchain data payloads (BlockScopedData), or some other trust-minimized payment mechanism.
+
+
+## Pricing
+
+Pricing is currently outside of the scope of this proposal. The protocol (contracts) itself does not provide pricing mechanisms, so this responsibility is passed to the Payment Gateway.
+
+The Payment Gateway MUST transfer funds to the Indexer according to a Pricing Period. Ideally this period is the shortest possible, to reduce counterparty risks.
+
+
+### Current implementation and future opportunities
+
+THIS SECTION IS NON-NORMATIVE
+
+Current pricing is set at a fixed price, defined on The Graph Market (https://thegraph.market). Consumption is measured in terms of Terabytes read from the backing stores and/or Egress bytes. Fees are cumulated during a period of time, until payment is settled. This time period will start at 1 month and be adjusted to become lower and lower as the system matures, to address counterparty risks.
+
+Although initial integration starts with a simple pricing strategy, the integration of Substreams on The Graph Network by no means limits the future opportunities to adjust and augment the business models and pricing strategies.
+
+Future GIPs will be proposed to enable more complex pricing strategies, such as pay-per-cpu-time, pay-per-storage, pay-per-query, pay-per-block, pay-per-attestation, etc.
+
+
+### Curation
+
+THIS SECTION IS NON-NORMATIVE
+
+Curation can happen on the published Subgraph manifest above, although a nonce may be added to avoid paying curation fees. Because there are no indexer rewards, curation is not adding value, therefore it is reasonable for participants to want to avoid it.
```

Comment:
I think you meant _Substreams_ here, and not _Subgraph_. Also, a nitpick, but we say _indexing_ rewards, is the right terminology and not _indexer rewards_ ;) 

IIRC, we discussed 2 options to avoid curation fees: 

1. using a single fixed deployment ID (manifest) where curators would share the 10% fee
2. creating new deployment IDs (I believe by adding nonces as mentioned here) before collecting payments

It seems like you're proposing #2, but that's still vulnerable to a front-running attack (like a bot adding curation signal right before the new allocations). But I don't _think_ we had agreed on a particular best solution. 

Also, I assume the "payment gateway" would monitor such curation and handle allocations accordingly before collecting payments, right?